spec/loophp/collection/CollectionSpec.php

```diff
@@ -226,9 +230,18 @@ public function it_can_asyncMap(): void
             return $v * 2;
         };
 
-        $this::fromIterable(['c' => 3, 'b' => 2, 'a' => 1])
-            ->asyncMap($callback1, $callback2)
-            ->shouldIterateAs(['a' => 2, 'b' => 4, 'c' => 6]);
+        $this->beConstructedThrough('fromIterable', [['c' => 3, 'b' => 2, 'a' => 1]]);
+
+        if (PHP_VERSION_ID >= self::PHP_VERSION_8) {
+            $this
+                ->asyncMap($callback1, $callback2)
+                ->shouldThrow(Error::class)
+                ->during('all');
+        } else {
+            $this
```

Comment:
I've modified this test case to show that an `Error` is thrown for `asyncMap` in PHP > 8.0.

That way the test still passes and we have this visibility

Reply:
I did [this](https://github.com/friends-of-phpspec/phpspec-requires) once upon a time, maybe we should see if it would be a better choice?

Reply:
that would be nice but even better actually I was going to ask if we should just throw an intentional exception from `asyncMap` itself if it's called in an environment running php 8. What do you think? @drupol 

That way we make it clear it's unsupported for now and we can add this info the [documentation](https://loophp-collection.readthedocs.io/en/latest/pages/api.html#asyncmap) as well

Reply:
Yes this is a good idea.

Reply:
updated 😁

Reply:
Saw it, seems like it is what we need. Nice!